### DISCUSSION TITLE
Feature Request: Add a `latest` tag

### DISCUSSION THREAD

**Discussion by abhillman** [duckdb/duckdb#9204] [Ideas]
When fetching duckdb programmatically, it can be reasonable to want to fetch the _latest_ version, which is not currently possible in a trivial manner.

Alternatives:
* a `latest-release` tag
* an http endpoint that returns plain text with the latest release number
* a documented, small bash statement that computes the latest tag and assigns it to an environment variable

Reasons against:
* allowing for a _latest_ tag can, in some cases, be harmful, as it can lead to non- predictable/stable/deterministic behavior, especially if breaking changes are introduced; i.e. _pinning_ is generally a good practice

Another consideration:
* one of the above approaches _might_ be useful to linux package manager maintainers

**Comment by abhillman** [comment]
Regarding the third alternative:

```bash
% export DUCKDB_LATEST_VERSION=$(git -c 'versionsort.suffix=-' ls-remote --tags --sort='-v:refname' https://github.com/duckdb/duckdb.git v\* |\
        head -n 1 |\
        awk '{ split($2, arr, "/"); print arr[3] }')
% echo $DUCKDB_LATEST_VERSION 
v0.9.0
```

Note: `awk` was chosen in lieu of `cut` due to differing semantics in the respective AT&T UNIX and GNU Coreutils versions of `cut`; the former ships with Mac OS X and the latter typically shipping with GNU/Linux

**Comment by Mause** [comment]
```sh
$ git describe --tags --abbrev=0
v0.9.0
```

Is what we use internally - does that give you what you need?
